internal/source/upload.go

```diff
@@ -9,37 +9,38 @@ import (
 	"github.com/nlepage/go-tarfs"
 
 	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+	pkgsource "github.com/operator-framework/rukpak/pkg/source"
 )
 
 // Upload is a bundle source that sources bundles from the rukpak upload service.
 type Upload struct {
-	baseDownloadURL string
-	bearerToken     string
-	client          http.Client
+	BaseDownloadURL string
+	BearerToken     string
+	Client          http.Client
```

Comment:
I had a similar comment in another one of the exporting an internal library PRs: does it make sense to create a constructor function here for clients that need access to the upload struct? It would allow us to keep these fields unexported and the fields would be immutable after assignment.

Reply:
I believe that the `upload.go` is under the `internal` directory so they are not exported to the other repository.  It's a good idea to make it  immutable.   I'll add the constructor.